dnadoh/synth_person.py

```diff
@@ -0,0 +1,77 @@
+"""Synthesize person data."""
+
+import random
+from datetime import date
+
+from faker import Faker
+from pydantic import BaseModel
+
+NORMAL_RANGE = 4.0  # Max. number of std. dev. allowed from mean
+
+
+class PersonParams(BaseModel):
+    """Params for person synthesis."""
+
+    min_dob: date = date(1920, 1, 1)
+    max_dob: date = date(2021, 1, 2)
+    infant_weight: float = 9.0
+    adult_weight: float = 80.0
+    std_weight: float = 0.25
+
+
+# Fake object.
+_FAKER = None
+
+
+def get_faker(seed=12345):
+    """Get (construct) global faker."""
+    global _FAKER
+    if _FAKER is None:
+        _FAKER = Faker(seed)
+    return _FAKER
+
+
+def rand_date_of_birth(params):
+    """Create a random date of birth."""
+    return get_faker().date_between_dates(
+        date_start=params.min_dob, date_end=params.max_dob
+    )
+
+
+def rand_weight(params, age):
+    """Create random weight based on age."""
+    return _piecewise(
+        lambda age: _normal_if(
+            age <= 25,
+            params.infant_weight
+            + (params.adult_weight - params.infant_weight) * (age / 25),
+            params.std_weight,
+        ),
+        lambda age: _normal_if(25 < age, params.adult_weight, params.std_weight),
+        age=age,
+    )
+
+
+def _normal_if(cond, mean, std, lowest=None):
+    """Sample from normal distribution if condition holds."""
+    if not cond:
+        return None
+    std = std * mean
```

Comment:
should probably rename the parameter `std_fraction` to avoid confusion between relative and absolute value.